kotlin/export-tiles/src/main/java/com/example/exporttiles/MainActivity.kt

```diff
@@ -0,0 +1,240 @@
+import android.graphics.Point
+import android.graphics.PorterDuff
+import android.graphics.PorterDuffColorFilter
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.esri.arcgisruntime.concurrent.Job
+import com.esri.arcgisruntime.concurrent.ListenableFuture
+import com.esri.arcgisruntime.data.TileCache
+import com.esri.arcgisruntime.geometry.Envelope
+import com.esri.arcgisruntime.layers.ArcGISTiledLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheJob
+import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheParameters
+import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheTask
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.dialog_layout.*
+import java.io.File
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG: String = MainActivity::class.java.simpleName
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create an ArcGISTiledLayer to use as the basemap
+    val tiledLayer = ArcGISTiledLayer(getString(R.string.world_street_map))
+    val map = ArcGISMap().apply {
+      basemap = Basemap(tiledLayer)
+      minScale = 10000000.0
+    }
+    // create a graphics overlay
+    val graphicsOverlay = GraphicsOverlay()
+    mapView.apply {
+      // set the map to the map view
+      this.map = map
+      setViewpoint(Viewpoint(51.5, 0.0, 10000000.0))
+      // add the graphics overlay to the map view
+      graphicsOverlays.add(graphicsOverlay)
+    }
+
+    // create a graphic to show a red box around the tiles we want to download
+    val downloadArea = Graphic()
+    graphicsOverlay.graphics.add(downloadArea)
+    val simpleLineSymbol = SimpleLineSymbol(SimpleLineSymbol.Style.SOLID, Color.RED, 2f)
+    downloadArea.symbol = simpleLineSymbol
+
+    // update the box whenever the viewpoint changes
+    mapView.addViewpointChangedListener {
+      if (mapView.map.loadStatus == LoadStatus.LOADED) {
+        // upper left corner of the downloaded tile cache area
+        val minScreenPoint = Point(150, 175)
+        // lower right corner of the downloaded tile cache area
+        val maxScreenPoint = Point(
+          mapView.getWidth() - 150,
+          mapView.getHeight() - 250
+        )
+        // convert screen points to map points
+        val minPoint: com.esri.arcgisruntime.geometry.Point? =
```

Comment:
I usually try to import the Runtime Point preferentially over the Android Point so that Runtime Point doesn't look so out of place in the code